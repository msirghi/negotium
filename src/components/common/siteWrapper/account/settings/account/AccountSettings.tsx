import { Box } from '@mui/system';
import { useSelector } from 'react-redux';
import { RootState } from '../../../../../../redux/store';
import { AccountInfo } from '../../../../../../common/types/account.types';
import { Avatar, Button, TextField } from '@mui/material';
import { deepOrange } from '@mui/material/colors';
import { makeStyles } from '@mui/styles';
import { Row } from '../../../../utilities/row/Row';
import colors from '../../../../../../common/styles/colors';
import { useCommonStyles } from '../../styles';
import { useTranslation } from 'next-i18next';
import {useIsMobile} from "../../../../../../common/hooks/common/useIsMobile";

const useStyles = makeStyles({
  photoContent: {
    marginLeft: 10,
  },
  photoHelperText: {
    color: colors.greys['500'],
    fontSize: 12,
    marginTop: 5,
  },
});

export const AccountSettings = () => {
  const accountInfo: AccountInfo = useSelector(
    (state: RootState) => state.account.info
  );
  const isMobile = useIsMobile();
  const classes = useStyles();
  const commonClasses = useCommonStyles();
  const { t } = useTranslation('settings');

  return (
    <Box>
      <div>
        <Box className={commonClasses.sectionTitle}>Photo</Box>
        <Box className={commonClasses.sectionBody}>
          <Row alignVerticalCenter>
            <Avatar sx={{ bgcolor: deepOrange[500], width: 56, height: 56 }}>
              {accountInfo.name[0]}
            </Avatar>
            <Box className={classes.photoContent}>
              <Button size={'small'} variant={'outlined'}>
                {t('photo.upload')}
              </Button>
              <Box className={classes.photoHelperText}>
                {t('photo.helperText')}
              </Box>
            </Box>
          </Row>
        </Box>

        <Box className={commonClasses.sectionTitle}>Name</Box>
        <Box className={commonClasses.sectionBody}>
          <TextField
            id={'name-field'}
            size={'small'}
            placeholder={'Name'}
            value={accountInfo.name}
            fullWidth={isMobile}
          />
        </Box>

        <Box className={commonClasses.sectionTitle}>Email</Box>
        <Box className={commonClasses.sectionBody}>
          <Box>
            <span>{accountInfo.email}</span>
          </Box>
          <Button size={'small'} variant={'outlined'} sx={{ marginTop: 1 }} fullWidth={isMobile}>
            Change email
          </Button>
        </Box>

        <Box className={commonClasses.sectionTitle}>Password</Box>
        <Box className={commonClasses.sectionBody}>
          <Button size={'small'} variant={'outlined'} sx={{ marginTop: 1 }} fullWidth={isMobile}>
            Change Password
          </Button>
        </Box>
      </div>
    </Box>
  );
};
