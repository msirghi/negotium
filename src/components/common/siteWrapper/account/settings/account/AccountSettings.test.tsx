import { accountInfoMock } from '../../../../../../common/tests/mockData/account-mock';
import renderer from 'react-test-renderer';
import { MockReduxProvider } from '../../../../../../common/tests/TestUtils';
import { AccountSettings } from './AccountSettings';
import { mount } from 'enzyme';

describe('AccountSettings', () => {
  const defaultProps = {};
  const reduxState = {
    account: {
      info: accountInfoMock,
    },
  };

  it('should match the snapshot', () => {
    const tree = renderer.create(
      <MockReduxProvider reduxStore={reduxState}>
        <AccountSettings {...defaultProps} />
      </MockReduxProvider>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render input with the default value', () => {
    const wrapper = mount(
      <MockReduxProvider reduxStore={reduxState}>
        <AccountSettings {...defaultProps} />
      </MockReduxProvider>
    );
    const nameField = wrapper.find('#name-field').at(0);
    expect(nameField.props().value).toEqual(accountInfoMock.name);
  });
});
