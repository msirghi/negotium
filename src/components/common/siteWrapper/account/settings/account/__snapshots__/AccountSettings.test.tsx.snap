// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSettings should match the snapshot 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="makeStyles-sectionTitle-4 MuiBox-root css-0"
  >
    photo.title
  </div>
  <div
    className="makeStyles-sectionBody-5 MuiBox-root css-0"
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "width": "initial",
        }
      }
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-s0wybj-MuiAvatar-root"
      >
        n
      </div>
      <div
        className="makeStyles-photoContent-1 MuiBox-root css-0"
      >
        <button
          className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          photo.upload
        </button>
        <div
          className="makeStyles-photoHelperText-2 MuiBox-root css-0"
        >
          photo.helperText
        </div>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-sectionTitle-4 MuiBox-root css-0"
  >
    name
  </div>
  <div
    className="makeStyles-sectionBody-5 MuiBox-root css-0"
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          data-testid="name-field"
          disabled={false}
          id="name-field"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="name"
          required={false}
          type="text"
          value="name"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1a1ihhj"
          >
            <span
              className="notranslate"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-sectionTitle-4 MuiBox-root css-0"
  >
    email
  </div>
  <div
    className="makeStyles-sectionBody-5 MuiBox-root css-0"
  >
    <div
      className="MuiBox-root css-0"
    >
      <span>
        mail@yahoo.com
      </span>
    </div>
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root css-zwh7hc-MuiButtonBase-root-MuiButton-root"
      data-testid="email-change-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      changeEmail
    </button>
  </div>
  <div
    className="makeStyles-sectionTitle-4 MuiBox-root css-0"
  >
    password
  </div>
  <div
    className="makeStyles-sectionBody-5 MuiBox-root css-0"
  >
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root css-zwh7hc-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      changePassword
    </button>
  </div>
</div>
`;
